### ISSUE TITLE
Using squash and merge instead of plain merge for pull requests

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#2499]
I'd like to propose changing our default merge strategy for pull requests to [squash and merge](https://learn.microsoft.com/en-us/azure/devops/repos/git/merging-with-squash?view=azure-devops):
> Squash merging is a merge option that allows you to condense the Git history of topic branches when you complete a pull request. Instead of each commit on the topic branch being added to the history of the default branch, a squash merge adds all the file changes to a single new commit on the default branch.

[Much](https://www.lloydatkinson.net/posts/2022/should-you-squash-merge-or-merge-commit/) [has](https://betterprogramming.pub/git-merge-squash-rebase-or-pull-what-to-choose-50b331d3e7c1) [been](https://rietta.com/blog/github-merge-types/) [written](https://blog.mergify.com/what-is-the-difference-between-a-merge-commit-a-squash/) about the tradeoffs between the various merge strategies. Generally, the benefit of squash and merge is that [it](https://learn.microsoft.com/en-us/azure/devops/repos/git/merging-with-squash?view=azure-devops):
> keeps your default branch histories clean and easy to follow without demanding any workflow changes on your team. Contributors to the topic branch work how they want in the topic branch, and the default branches keep a linear history through the use of squash merges. The commit history of a `main` branch updated with squash merges has one commit for each merged branch. You can step through this history to find out exactly when work was done.

The downside is that you lose the detailed history of the feature branch. Personally, I've never had a desire or a good reason to dig through individual commits on a branch that was already merged, so to me the benefits outweigh this downside.

@openmc-dev/technical-committee

**Comment by gridley** [comment]
Yeah, we absolutely should be squashing every PR, IMO. Most of the time the commits are changes that just modify lines of code that were already changed when incorporating review comments. There is no reason at all to keep this kind of thing in the history. Ideally, we could check out any commit and have it pass CI.

**Comment by eepeterson** [comment]
+1 fully agree

**Comment by pshriwise** [comment]
+1 📈

**Comment by shimwell** [comment]
+1 :flatbread:

**Comment by paulromano** [comment]
Seems like everyone is in favor of this change, so I've gone ahead and made it the default. I'll still keep this open for a little bit longer in case anyone else wants to dissent.
